sklearn/gaussian_process/kernels.py

```diff
@@ -313,7 +314,7 @@ def __pow__(self, b):
         return Exponentiation(self, b)
 
     def __eq__(self, b):
-        if type(self) != type(b):
```

Comment:
I think this condition is more correct. Did you have a strict reason to change it? Using `is` would be better than `!=`.